src/composeApolloProvider.tsx

```diff
@@ -0,0 +1,72 @@
+/// <reference path="../typings/main.d.ts" />
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import {
+    Component,
+    PropTypes,
+} from 'react';
+
+import {
+    Store,
+} from 'redux';
+
+
+import ApolloClient from 'apollo-client';
+
+declare interface ProviderProps {
+    store?: Store<any>;
+    client: ApolloClient;
+}
+
+export default function composeApolloProvider<T>(Provider: React.ComponentClass<any>): any {
```

Comment:
I guess I feel like it just adds an extra line of code, but doesn't really add much clarity since you still don't know what it's doing with the `Provider`. If we wanted people to have the ultimate clarity about what the provider is doing we could just ask them to use two providers:

```js
<Provider>
   <ApolloProvider>
      ...
```